rhel/vex/parser_test.go

```diff
@@ -373,11 +374,198 @@ func TestParse(t *testing.T) {
+				Name:      "buildah-debugsource",
+				Version:   "1.24.6-5.module+el8.8.0+18083+cd85596b",
+				Qualifiers: packageurl.QualifiersFromMap(map[string]string{
+					"arch":  "ppc64le",
+					"epoch": "1",
+				}),
+			},
+			want: "1:1.24.6-5.module+el8.8.0+18083+cd85596b",
+		},
+		{
+			name: "rpm_without_epoch",
+			purl: packageurl.PackageURL{
+				Type:      packageurl.TypeRPM,
+				Namespace: "redhat",
+				Name:      "buildah-debugsource",
+				Version:   "1.24.6-5.module+el8.8.0+18083+cd85596b",
+				Qualifiers: packageurl.QualifiersFromMap(map[string]string{
+					"arch": "ppc64le",
+				}),
+			},
+			want: "0:1.24.6-5.module+el8.8.0+18083+cd85596b",
+		},
+		{
+			name: "oci_with_tag",
+			purl: packageurl.PackageURL{
+				Type:      packageurl.TypeOCI,
+				Namespace: "",
+				Name:      "keepalived-rhel9",
+				Version:   "sha256:36abd2b22ebabea813c5afde35b0b80a200056f811267e89f0270da9155b1a22",
+				Qualifiers: packageurl.QualifiersFromMap(map[string]string{
+					"arch":           "ppc64le",
+					"repository_url": "registry.redhat.io/rhceph/keepalived-rhel9",
+					"tag":            "2.2.4-3",
+				}),
+			},
+			want: "2.2.4-3",
+		},
+		{
+			name: "oci_without_tag",
+			purl: packageurl.PackageURL{
+				Type:      packageurl.TypeOCI,
+				Namespace: "",
+				Name:      "keepalived-rhel9",
+				Version:   "sha256:36abd2b22ebabea813c5afde35b0b80a200056f811267e89f0270da9155b1a22",
+				Qualifiers: packageurl.QualifiersFromMap(map[string]string{
+					"arch":           "ppc64le",
+					"repository_url": "registry.redhat.io/rhceph/keepalived-rhel9",
+				}),
+			},
+			expectedErr: true,
+		},
+
+		{
+			name: "unsupported_type",
+			purl: packageurl.PackageURL{
+				Type:      packageurl.TypeApk,
+				Namespace: "",
+				Name:      "nice APK",
+				Version:   "v1.1.1",
+				Qualifiers: packageurl.QualifiersFromMap(map[string]string{
+					"arch": "ppc64le",
+				}),
+			},
+			expectedErr: true,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			v, err := extractFixedInVersion(tc.purl)
+			if !errors.Is(err, nil) && !tc.expectedErr {
+				t.Fatalf("expected no err but got %v", err)
+			}
+			if errors.Is(err, nil) && tc.expectedErr {
+				t.Fatal("expected err but got none")
+			}
+			if v != tc.want {
+				t.Fatalf("expected version %v but got %v", tc.want, v)
+			}
+		})
+	}
+}
+
+func TestExtractPackageName(t *testing.T) {
+	testcases := []struct {
+		name        string
+		purl        packageurl.PackageURL
+		expectedErr bool
+		want        string
+	}{
+		{
+			name: "rpm_simple",
+			purl: packageurl.PackageURL{
+				Type:      packageurl.TypeRPM,
+				Namespace: "redhat",
+				Name:      "buildah-debugsource",
+				Version:   "1.24.6-5.module+el8.8.0+18083+cd85596b",
+				Qualifiers: packageurl.QualifiersFromMap(map[string]string{
+					"arch":  "ppc64le",
+					"epoch": "1",
+				}),
+			},
+			want: "buildah-debugsource",
+		},
+		{
+			name: "oci_with_repository_url",
```

Comment:
want to add one more for when `repository_url` exists, but `Name` has a namespace?

Reply:
Done